### PR TITLE
feature/versioning-for-overrides-BZ-6949

### PR DESCRIPTION
**Pull request by tiandavis**
## Summary - [BZ-6949](https://thirdiron.atlassian.net/browse/BZ-6949)

Enables customers to override default text by explicitly declaring `version` `2` in the Summon integration script. In the mean time, we provide helpful implicit defaults.

## Description

Example Integration Script
image

## Deploy Precautions

- None

### CODE REVIEW

**Review by karlbecker — CHANGES_REQUESTED** [review]
Looking pretty good to me @tiandavis  🙌   Setting this to request changes since I think there's an example in the README we need to change, and a couple tests to add, including perhaps one additional error-related scenario to handle. And I think at least one of those could be punted to another PR, too, but let me know what you think.

Thanks Tian! Almost there 💪

File: src/summon/browzine-summon-adapter.js [hunk 1]
```diff
@@ -470,28 +496,32 @@ browzine.summon = (function() {
...
+  function directToPDFTemplate(directToPDFUrl, articleRetractionUrl) {
     var pdfIcon = getPdfIconSvg();
     var pdfIconWidth = "13";
-    var articlePDFDownloadWording = browzine.articlePDFDownloadWording || browzine.summonArticlePDFDownloadWording || "Article PDF";
-    var articlePDFDownloadLinkText = browzine.articlePDFDownloadLinkText || browzine.summonArticlePDFDownloadLinkText || "Download Now";
+
+    var articlePDFDownloadWording = "View Now";
+    var articlePDFDownloadLinkText = "PDF";
+
+    if (browzine.version >= 2) {
```
[the remark this answers is not in the document]
Comment (tiandavis) [reply src/summon/browzine-summon-adapter.js:508]: @karlbecker Don't overthink this one. We should be good. Checkout this test:
image

If we really want to ensure number to number comparison, we can force a typecast with the `+` symbol:
```
if (+browzine.version >= 2) {
```

File: tests/unit/models/summon.js [hunk 2]
```diff
@@ -77,7 +77,29 @@ describe("Summon Model >", function() {
...
+          issns: ["0028-4793"]
+        };
+
+        documentSummary = $compile(documentSummary)($scope);
+      });
+
+      var primaryColor = summon.getPrimaryColor(documentSummary);
+
+      expect(primaryColor).toBeDefined();
+      expect(primaryColor).toEqual("rgb(38, 89, 171)");
```
[the remark this answers is not in the document]
Comment (tiandavis) [reply tests/unit/models/summon.js:98]: Hopefully, we'll get to use it. Maybe John will know where the pdf color comes from.

**Comment by tiandavis** [comment]
@karlbecker Should be all set now! 👍 🥃

File: src/summon/browzine-summon-adapter.js [hunk 3]
```diff
@@ -505,7 +505,7 @@ browzine.summon = (function() {
     var articlePDFDownloadWording = "View Now";
     var articlePDFDownloadLinkText = "PDF";
 
-    if (browzine.version >= 2) {
+    if (+browzine.version >= 2) {
```
Comment (karlbecker) [inline src/summon/browzine-summon-adapter.js:508]: TIL about the unary `+` operator: https://medium.com/@nikjohn/cast-to-number-in-javascript-using-the-unary-operator-f4ca67c792ce

I also checked to see what happens if `version` was set to something like "2a", "tbd", etc., and I see `+browzine.version >= 2` should evaluate to falsy:

CleanShot 2021-12-13 at 08 59 53@2x

Only thing I can think of to maybe add a test - which we could do after merging this PR - is writing a test to define the behavior when `version` is set to an alphanumeric like that or other invalid value.

**Review by karlbecker — APPROVED** [review]
Looking pretty good to me @tiandavis 👍 I think the only change I think would be nice to get in before merging (or shortly after merging this) is to clarify the precondition on a few tests.  What do you think?

Other than that, I think we're good to go!  And we could even merge it now so testing can start, since we can make another PR after this to clarify those test descriptions.

Thanks for the hard work on getting this across the finish line in time for the launch of this probably late tomorrow Tian! 🏆

File: src/summon/browzine-summon-adapter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (karlbecker) [reply src/summon/browzine-summon-adapter.js]: Oh lol @tiandavis , I didn't even see this comment until after I did the review - and we both checked for the string version comparison behavior using the JS console in the browser, haha.  Great minds 😊 

Looks great, thanks for teaching me about `+` 👍

**Comment by karlbecker** [comment]
@JMalone29 is ready to test, so since the functionality seems to be all there @tiandavis , I'm going to merge this in to let the testing begin!

challenge-accepted-let-the-challenge-commence

**Pull request merged**